scanomatic/image_analysis/image_fixture.py

```diff
@@ -9,112 +8,103 @@
 
 from . import image_basics
 from .image_basics import load_image_to_numpy
+from .exceptions import FixtureImageError
+
+
+_logger = get_logger("Resource Image Analysis")
 
 
 class FixtureImage:
     def __init__(
         self,
-        path=None,
-        image=None,
-        pattern_image_path=None,
-        scale=1.0,
-        resource_paths=None,
+        image: np.ndarray,
+        pattern_image: np.ndarray,
+        scale: float,
     ):
-        self._path = path
-        self._img = None
-        self._pattern_img = None
-        self._load_error = None
+        self._img = image
+        self._pattern_img = pattern_image
         self._transformed = False
         self._conversion_factor = 1.0 / scale
-        self._logger = get_logger("Resource Image Analysis")
 
-        self._logger.info(
-            "Analysing image {0} using pattern file {1} and scale {2}".format(
-                path if path else (
-                    image.shape if image is not None else "NO IMAGE"
-                ),
+    @classmethod
+    def from_image(
+        cls,
+        image: np.ndarray,
+        pattern_image_path: str,
+        scale: float = 1.0,
+    ) -> "FixtureImage":
+        if len(image.shape) > 2:
+            image = image[:, :, 0]
+
+        try:
+            pattern_image = load_image_to_numpy(
                 pattern_image_path,
-                scale
-            ),
-        )
+                dtype=np.uint8,
+            )
+        except IOError:
+            msg = f"Could not open orientation guide image at {pattern_image_path}"  # noqa: E501
+            _logger.error(msg)
+            raise FixtureImageError(msg)
+
+        if len(pattern_image.shape) > 2:
+            pattern_image = pattern_image[:, :, 0]
+
+        return cls(image, pattern_image, scale)
+
+    @classmethod
+    def from_image_path(
+        cls,
+        image_path: str,
+        pattern_image_path: str,
+        scale: float = 1.0,
+    ) -> "FixtureImage":
+        try:
+            image = load_image_to_numpy(image_path, dtype=np.uint8)
 
-        if (
-            os.path.isfile(pattern_image_path) is False
-            and resource_paths is not None
-        ):
-            pattern_image_path = os.path.join(
-                resource_paths.images,
-                os.path.basename(pattern_image_path),
```

Comment:
I didn't understand what this was for and it doesn't seem to be used so I just scrapped it. I'll add it back in if anyone objects.